kubernetes/utils.go

```diff
@@ -1020,16 +1020,36 @@ func normalizeCPUToMilliCores(cpu string) (int64, error) {
 
 // normalizeMemoryToBytes converts memory quantities to bytes, rounding up if necessary.
 func normalizeMemoryToBytes(memory string) (int64, error) {
-	// Set default value to arg value and unit to Bytes
+	// Handle scientific notation by trying to parse the full string as a float first
+	if value, err := strconv.ParseFloat(memory, 64); err == nil {
+		// If it parses successfully as a pure number (including scientific notation), treat as bytes
+		return int64(math.Ceil(value)), nil
+	}
+
+	// Find the boundary between number and unit more carefully
+	// Look for the start of alphabetic characters that don't include 'e' or 'E' (for scientific notation)
 	valuePart := memory
 	unitPart := "B"
+
 	for i, r := range memory {
-		if r < '0' || r > '9' {
+		// If we find an alphabetic character that's not 'e' or 'E', or if we find 'e'/'E'
+		// followed by an alphabetic character (not a digit or +/-), then we've found the unit
+		if (r >= 'A' && r <= 'Z' && r != 'E') || (r >= 'a' && r <= 'z' && r != 'e') {
 			valuePart = memory[:i]
 			unitPart = memory[i:]
 			break
 		}
+		// Handle the case where 'e' or 'E' is followed by a letter (not +/- or digit)
+		if (r == 'e' || r == 'E') && i+1 < len(memory) {
+			next := memory[i+1]
+			if (next >= 'A' && next <= 'Z') || (next >= 'a' && next <= 'z') {
+				valuePart = memory[:i]
+				unitPart = memory[i:]
+				break
```

Comment:
Do we no longer need the `if r < '0' || r > '9'` check? I do not think it is handled in the code anymore.

Reply:
Hi @misraved, You're right that the original `r < '0' || r > '9'` check was used to detect the first non-digit character to separate the value and unit.

In the updated logic, that check has been replaced by a more explicit and accurate condition that:

* Identifies the first alphabetic character (excluding `'e'`/`'E'`)
* Handles the special case where `'e'` or `'E'` is followed by another alphabetic character, which isn't valid scientific notation

So functionally, the new implementation **covers and refines** the original intent of the `r < '0' || r > '9'` check. 👍

Let me know if you think we should make that even more explicit!